core/common/src/MapImagerySettings.ts

```diff
@@ -22,31 +22,214 @@ export type BaseLayerProps = MapLayerProps | ColorDefProps;
+    switch (this.mapType) {
+      case BackgroundMapType.Aerial:
+        return "Aerial Imagery";
+      default:
+      case BackgroundMapType.Hybrid:
+        return "Aerial Imagery with labels";
+      case BackgroundMapType.Street:
+        return "Streets";
+    }
+  }
+
+  public toMapSettings(): MapLayerSettings {
+    let formatId: string, url: string, name: string;
+    switch (this.providerName) {
+      case "BingProvider":
+      default:
+        formatId = "BingMaps";
+
+        let imagerySet;
+        switch (this.mapType) {
+          case BackgroundMapType.Street:
+            imagerySet = "Road";
+            break;
+          case BackgroundMapType.Aerial:
+            imagerySet = "Aerial";
+            break;
+          case BackgroundMapType.Hybrid:
+          default:
+            imagerySet = "AerialWithLabels";
+            break;
+        }
+        name = `Bing Maps: ${this.mapTypeName()}`;
+        url = `https://dev.virtualearth.net/REST/v1/Imagery/Metadata/${imagerySet}?o=json&incl=ImageryProviders&key={bingKey}`;
+        break;
+
+      case "MapBoxProvider":
+        formatId = "MapboxImagery";
+        name = `MapBox: ${this.mapTypeName()}`;
+        switch (this.mapType) {
+          case BackgroundMapType.Street:
+            url = "https://api.mapbox.com/v4/mapbox.streets/";
+            break;
+          case BackgroundMapType.Aerial:
+            url = "https://api.mapbox.com/v4/mapbox.satellite/";
+            break;
+          case BackgroundMapType.Hybrid:
+            url = "https://api.mapbox.com/v4/mapbox.streets-satellite/";
+            break;
+        }
+        break;
+    }
+    return MapLayerSettings.fromJSON({ name, formatId, url, transparentBackground: false, isBase: true })!;
+  }
+
+  public clone(changedProps: BackgroundMapProviderProps): BackgroundMapProvider {
+    if (undefined === changedProps)
+      return this;
+
+    const props = {
+      name: undefined !== changedProps.providerName ? changedProps.providerName : this.providerName,
+      providerData: {mapType: undefined !== changedProps.providerData?.mapType ? changedProps.providerData.mapType : this.mapType},
+
+    };
+    const clone = BackgroundMapProvider.fromJSON(props)!;
+
+    return clone;
+  }
+
+  /** Construct from JSON, performing validation and applying default values for undefined fields. */
+  public static fromJSON(props: BackgroundMapProviderProps) {
+    return new BackgroundMapProvider(props);
+  }
+
+  public static isMatchingProps(props: any): boolean {
+    return props?.hasOwnProperty("providerName");
+  }
 }
 
 /** Normalized representation of base layer properties -- these can be represented by either a full map layer or a simple color.
  * @beta
  */
-export type BaseLayerSettings = MapLayerSettings | ColorDef;
+export type BaseLayerContent = MapLayerSettings | ColorDef | BackgroundMapProvider;
+export type BaseLayerContentProps = MapLayerProps | ColorDefProps | BackgroundMapProviderProps;
```

Comment:
New 'content' type that specify the base layer as either a Map layer definition, a color or a BackgroundMapProvider